Canny source doc (#797)
rudderlabs/rudderstack-docs: docs/sources/cloud-apps/canny.mdx

1. Go to your [RudderStack dashboard](https://app.rudderstack.com/) and click on **Add Source**. Then, select **Canny** from the list of **Event Stream** sources, as shown:

<img src="../../assets/event-stream-sources/canny-1.png" alt="Canny source" />

2. Assign a name to your source and click on **Next**.
3. The Canny source is now configured. Note the **Webhook URL** in the **Settings** tab.

<img src="../../assets/event-stream-sources/canny-2.png" alt="Canny source webhook URL" />

4. Go to your Canny account and navigate to **Settings** > **API & Webhooks** under your profile.
5. Enter the **Webhook URL** under **Webhooks** section, as shown:

<img src="../../assets/event-stream-sources/webhook-url-canny.png" alt="Canny source webhook URL" />

6. Finally, click on **Add**. 

## Event transformation

RudderStack ingests the events from Canny after converting them into the RudderStack event format. It also maps the following properties from the Canny event payload to the RudderStack properties:

| Canny property  | RudderStack property| Notes |
| :------------------ | :------------- |:------------- |
| `created`  | `timestamp`,`originalTimestamp`    | - |

> | `object`        | `properties.objectType`        | -|

Reviewer: ```suggestion
| `objectType`        | `properties.objectType`        | -|
```

Reviewer: object-> properties mapping is missing

<img src="../../assets/event-stream-sources/webhook-url-canny.png" alt="Canny source webhook URL" />

6. Finally, click on **Add**. 

## Event transformation

RudderStack ingests the events from Canny after converting them into the RudderStack event format. It also maps the following properties from the Canny event payload to the RudderStack properties:

| Canny property  | RudderStack property| Notes |
| :------------------ | :------------- |:------------- |
| `created`  | `timestamp`,`originalTimestamp`    | - |
| `object`        | `properties`        | -|
| `objectType`        | `properties.objectType`        | -|
| `type`    | `event`  | -|
| `object.author.userId`     | `userId`   |- |
| sha256 hash of(`object.author.email`)        | `anonymousId` | Only if `userId` is not present |
| sha256 hash of(`object.voter.email`)        | `anonymousId` |Only if `userId` is not present AND `type`is `vote.created`/`vote.deleted`|
| `object.voter.userId`    | `userId`  | Only if `type`is `vote.created`/`vote.deleted` |
| `object.voter.id`        | `context.externalId[0].value` | Only if `type`is `vote.created`/`vote.deleted`|

> | `cannyId`    | `context.externalId[0].type`  |- |

Reviewer 2: 'cannyId' does not belong to Canny, we are assigning the type of externalId as `cannyUserId`.

---

> title: "Canny"

Reviewer 3: ```suggestion
title: "Canny Source"
```

Reviewer 3: To help with the search results.

Author: Should we add it in alias tag rather? search results will be catered, and consistency across all source docs will be maintained too. WDYT

Reviewer 3: Makes sense!

description: Step-by-step guide on ingesting your event data from Canny into RudderStack.
---

[Canny](https://canny.io/) is a customer feedback management tool which captures, organizes, and analyzes product feedback in one place to help you make informed product decisions.

This guide will help you set up Canny as a source in RudderStack.

## Getting started

Follow these steps to set up Canny as a source in the RudderStack dashboard:

1. Go to your [RudderStack dashboard](https://app.rudderstack.com/) and click on **Add Source**. Then, select **Canny** from the list of **Event Stream** sources, as shown:

<img src="../../assets/event-stream-sources/canny-1.png" alt="Canny source" />

2. Assign a name to your source and click on **Next**.
3. The Canny source is now configured. Note the **Webhook URL** in the **Settings** tab.

<img src="../../assets/event-stream-sources/canny-2.png" alt="Canny source webhook URL" />

> 4. Go to your Canny account and navigate to **Settings** > **API & Webhooks** under your profile.

Reviewer 3: ```suggestion
4. Go to your <a href="https://canny.io/login">Canny account</a> and navigate to **Settings** > **API & Webhooks** under your profile.
```

5. Enter the **Webhook URL** under **Webhooks** section, as shown:

<img src="../../assets/event-stream-sources/webhook-url-canny.png" alt="Canny source webhook URL" />

6. Finally, click on **Add**. 

## Event transformation

> RudderStack ingests the events from Canny after converting them into the RudderStack event format. It also maps the following properties from the Canny event payload to the RudderStack properties:

Reviewer 3: ```suggestion
RudderStack ingests the Canny events after converting them into the RudderStack event format. It also maps the following properties from the Canny event payload to the RudderStack properties:
```

| Canny property  | RudderStack property| Notes |
| :------------------ | :------------- |:------------- |
| `created`  | `timestamp`,`originalTimestamp`    | - |

> | `object`        | `properties`        | excluding `author`/`voter` from properties|

Reviewer 3: ```suggestion
| `object`        | `properties`        | RudderStack excludes `author`/`voter` from the properties. |
```

Reviewer 3: Please verify if this is accurate.

| `objectType`        | `properties.objectType`        | -|
| `type`    | `event`  | -|

> | `object.author`    | `context.traits`  | excluding `userId` and `id` |

Reviewer 3: ```suggestion
| `object.author`    | `context.traits`  | RudderStack excludes `userId` and `id`. |
```

| `object.author.userId`     | `userId`   |- |

> | sha256 hash of(`object.author.email`)        | `anonymousId` | Only if `userId` is not present |

Reviewer 3: ```suggestion
| SHA-256 hash of(`object.author.email`)        | `anonymousId` | Only if `userId` is not present |
```

Reviewer 3: Are the brackets required here? Just confirming.

> | sha256 hash of(`object.voter.email`)        | `anonymousId` |Only if `userId` is not present AND `type`is `vote.created`/`vote.deleted`|

Reviewer 3: ```suggestion
| sha256 hash of(`object.voter.email`)        | `anonymousId` |Only if `userId` is not present **and** `type`is `vote.created`/`vote.deleted`|
```

> | sha256 hash of(`object.voter.email`)        | `anonymousId` |Only if `userId` is not present AND `type`is `vote.created`/`vote.deleted`|

Reviewer 3: ```suggestion
| SHA-256 hash of(`object.voter.email`)        | `anonymousId` |Only if `userId` is not present AND `type`is `vote.created`/`vote.deleted`|
```

> | `object.voter.userId`    | `userId`  | Only if `type`is `vote.created`/`vote.deleted` |

Reviewer 3: ```suggestion
| `object.voter.userId`    | `userId`  | Only if `type` is `vote.created`/`vote.deleted` |
```

> | `object.voter.id`        | `context.externalId[0].value` | Only if `type`is `vote.created`/`vote.deleted`|

Reviewer 3: ```suggestion
| `object.voter.id`        | `context.externalId[0].value` | Only if `type` is `vote.created`/`vote.deleted`|
```

| `cannyUserId`    | `context.externalId[0].type`  |- |
| `object.author.id`     | `context.externalId[0].id`   |- |

> Canny supports the following events:

Reviewer 3: ```suggestion
Canny supports ingesting the following events:
```
